compiler/testData/diagnostics/tests/delegatedProperty/typeMismatchForGetReturnType.fir.kt

```diff
@@ -3,7 +3,7 @@
 
 import kotlin.reflect.KProperty
 
-val c: Int by Delegate()
```

Comment:
Rather I'm curious why we missed `DELEGATE_SPECIAL_FUNCTION_NONE_APPLICABLE`: https://github.com/JetBrains/kotlin/blob/ad9b962536e0d0c148baf884111c2c97c2bc8f2c/compiler/testData/diagnostics/tests/delegatedProperty/typeMismatchForGetReturnType.kt#L6

Reply:
I think FE1.0 adds a return type constraint to the inference system and FIR is not doing that. IMO it's a bug that FE1.0 is considering return type when resolving method calls. But we can ask JetBrains for confirmation.